scripts/Muon/GUI/Common/phase_table_widget/phase_table_presenter.py

```diff
@@ -237,3 +233,23 @@ def update_current_groups_list(self):
     def update_current_phase_tables(self):
         phase_table_list = self.context.phase_context.get_phase_table_list(self.context.data_context.instrument)
         self.view.set_phase_table_combo_box(phase_table_list)
+
+    def handle_first_good_data_changed(self):
+        self._validate_data_changed(self.view.first_good_time, "First Good Data")
+
+    def handle_last_good_data_changed(self):
+        self._validate_data_changed(self.view.last_good_time, "Last Good Data")
+
+    def _validate_data_changed(self, data, string):
+        first_good_time = self.context.phase_context.options_dict['first_good_time']
+        last_good_time = self.context.phase_context.options_dict['last_good_time']
```

Comment:
you probably want to get these values from the main context https://github.com/mantidproject/mantid/blob/master/scripts/Muon/GUI/Common/contexts/muon_context.py#L378 

At the moment it says last cannot be greater than 15, but I know the data is good till 32. You would probably need the largest first and smallest last to make sure they are always valid when multiple runs are loaded.